client/extensions/woocommerce/state/data-layer/ui/products/test/index.js

```diff
@@ -332,6 +334,127 @@ describe( 'handlers', () => {
 				variation: variationBlack,
 			} ) );
 		} );
+
+		it( 'should create variations for an existing product', () => {
+			const product1 = { id: 42, name: 'Product #1', type: 'variable' };
+			const variationBlack = { id: 202, sku: 'old-sku', attributes: [ { name: 'Color', options: 'Black' } ] };
+
+			const rootState = {
+				extensions: {
+					woocommerce: {
+						ui: {
+							products: {
+								123: {
+									edits: {
+									},
+									variations: {
+										edits: [
+											{
+												productId: 42,
+												creates: [
+													variationBlack,
+												]
+											}
+										]
+									}
+								},
+							},
+						},
+						sites: {
+							123: {
+								products: {
+									products: [
+										product1,
+									]
+								},
+								productVariations: { },
+							}
+						},
+					}
+				}
+			};
+
+			const productEdits = rootState.extensions.woocommerce.ui.products[ 123 ].edits;
+			const variationEdits = rootState.extensions.woocommerce.ui.products[ 123 ].variations.edits;
+			const actionList = makeProductActionList( rootState, 123, productEdits, variationEdits );
+
+			expect( actionList.nextSteps.length ).to.equal( 1 );
+
+			const dispatch = spy();
+
+			// Create the variation.
+			actionList.nextSteps[ 0 ].onStep( dispatch, actionList );
+
+			expect( dispatch ).to.have.been.calledWith( match( {
+				type: WOOCOMMERCE_PRODUCT_VARIATION_CREATE,
+				siteId: 123,
+				productId: 42,
+				variation: variationBlack,
+			} ) );
+		} );
+
+		it( 'should update a variation for an existing product', () => {
+			const product1 = { id: 42, name: 'Product #1', type: 'variable' };
+			const variationBlack = { id: 202, sku: 'old-sku', attributes: [ { name: 'Color', options: 'Black' } ] };
+			const variationBlackEdit = { id: 202, sku: 'new-sku' };
+
+			const rootState = {
```

Comment:
Yeah, I think this test file probably needs to have a separate fixture added. But I think I'd like to save that for post-release.